core/drivers/clk/clk.c

```diff
@@ -226,21 +322,37 @@ TEE_Result clk_set_rate(struct clk *clk, unsigned long rate)
 
 struct clk *clk_get_parent(struct clk *clk)
 {
-	return clk->parent;
+	if (is_clk_elt(clk))
+		return clk_to_clk_elt(clk)->parent;
+
+	return NULL;
+}
+
+size_t clk_get_num_parents(struct clk *clk)
+{
+	if (is_clk_elt(clk))
+		return clk_to_clk_elt(clk)->num_parents;
+
+	return 0;
 }
 
 static TEE_Result clk_get_parent_idx(struct clk *clk, struct clk *parent,
 				     size_t *pidx)
 {
 	size_t i = 0;
 
+	if (!is_clk_elt(clk))
```

Comment:
IMHO, this check is redundant since `clk_get_num_parents()` would return 0 anyway and thus `TEE_ERROR_BAD_PARAMETERS` would be returned at line 357.
I know it would display the error message below but anyway, calling this with a lw clock would be an error.

Reply:
yes, could assert here.